Week-6/Assignment-5/index.js

```diff
@@ -0,0 +1,51 @@
+const A1 = [5, 10, 3, 2, 50, 80], B1 = 78;
+const A2 = [-10, 20], B2 = 30;
+const A3 = [8, 0, 2, 0, 3, 9, 10, 50, 103], B3 = 103;
+const A4 = [8, 0, 2, 0, 3, 9, 10, 50, 103], B4 = -106;
+const A5 = [-104, 20], B5 = 20;
+
+/**
+ * @function validateInput checks the input for 1 <= N <= 105 -103 <= A[i] <= 103 -105 <= B <= 105
+ * @param {Array} arr an array
+ * @returns boolean
+ */
+const validateInput = (arr, diff) => Array.isArray(arr) && arr.length > 1 && arr.length < 105 &&
+    arr.every(el => Number.isInteger(el) && el >= -103 && el <= 103) &&
+    Number.isInteger(diff) && diff >= -105 && diff <= 105;
+
+/**
+ * @function pairWithDiff finds a pair with a given difference if available
+ * @summary Time Complexity - O(n^2), Space Complexity - O(1)
+ * @param {Array} arr an input array
+ * @param {Number} diff a difference to find a pair with
+ * @returns 1 or 0 based on if a pair with given difference is found
+ */
+const pairWithDiff = (arr, diff) => {
```

Comment:
You can sort the array in O(nlogn) and then traverse the array from first and last to keep track of difference and then update your index accordingly. This way your time complexity will be reduced.

Reply:
Thanks @sairamin, I've updated my code now, please review